docs/pages/develop/authentication.mdx

```diff
@@ -1,67 +1,288 @@
+
+If you're using Expo Router deep linking is handled automatically (for most cases). You usually don't need to configure anything extra to make magic links work properly, which makes this approach even easier to adopt. See [Linking into your app](/linking/into-your-app) to learn more.
+
+[React Navigation](https://reactnavigation.org/) also supports deep linking, but you will need to configure it manually. See their [Deep Linking guide](https://reactnavigation.org/docs/deep-linking/) for more details.
+
+</Collapsible>
+
+<Collapsible summary="One-Time Passcodes (OTP)">
+
+An alternative to magic links is sending a one-time passcode by email or SMS. Instead of clicking a link, the user copies the code and manually returns to the app to enter it. This must happen within a specific time window before the code expires.
+
+There's no deep linking involved here. The user stays in control of the flow and must return to the app themselves.
+
+Fortunately, newer versions of Android and iOS automatically detect passcodes in incoming messages. This enables autofill suggestions above the keyboard, allowing users to enter the code with a single tap. When this works, the experience is seamless.
+
+> **info** Magic links and passcodes are both valid authentication methods for App Store and Play Store review. You can submit your app with either of these methods as the only option and get approved, even before adding social or OAuth login options.
+
+</Collapsible>
+
+## OAuth 2.0
+
+To let your users log in using their existing accounts from services like Google, Apple, GitHub, and more, you can use OAuth 2.0.
+
+[OAuth 2.0](https://oauth.net/2) is a widely used, secure protocol that allows your app to access user information from another service, without needing to handle passwords. It allows your users to to log in with a single tap, which saves time, builds trust, and removes the need to manage passwords.
+
+> **info** OAuth flows can be complex. If you're looking for a simple integration, most providers offer SDKs and services that handle everything for you. You can learn more about these in the [Auth Solutions](#auth-solutions) section.
+
+If you are looking for full control or want to understand how OAuth works under the hood, keep reading. This section shows how to implement a complete OAuth flow yourself using Expo.
+
+### How OAuth Works
+
+OAuth works by introducing an authorization server that acts as a secure middleman. Instead of giving your app their password, users log in through this server and approve access to specific data (like their name or email). The server then issues a temporary code, which your app can exchange for a secure access token.
+
+<ContentSpotlight src="/static/images/develop/authentication/oauth-flow.png" />
+
+> **info** In the above diagram, the term "client" does not imply any particular implementation characteristics (for example, whether the application executes on a server, a desktop, mobile app, or other devices).
```

Comment:
```suggestion
> **info** In the above diagram, the term "client" does not imply any particular implementation characteristics (for example, whether the application executes on a server, a desktop, a mobile app, or other devices).
```

This feels like, it should the diagram's caption. You can use `caption` prop on the `ContentSpotlight`.